security/server/ipsec_vici.go

```diff
@@ -105,6 +105,74 @@ type rekey_connection struct {
 	Reauth     bool   `vici:"reauth"`
 }
 
+type list_sas struct {
+	Ike        string `vici:"ike"`
+	IkeId      string `vici:"ike"`
+	Child      string `vici:"child"`
+	ChildId    string `vici:"child"`
+	Noblock    string `vici:"noblock"`
+}
+
+type list_child_sa struct {
+	Protocol      string `vici:"protocol"`
+	Encap         string `vici:"encap"`
+	SpiIn         string `vici:"spi-in"`
+	SpiOut        string `vici:"spi-out"`
+	CpiIn         string `vici:"cpi-in"`
+	CpiOut        string `vici:"cpi-out"`
+	MarkIn        string `vici:"mark-in"`
+	MarkMaskIn    string `vici:"mark-mask-in"`
+	MarkOut       string `vici:"mark-out"`
+	MarkMaskOut   string `vici:"mark-mask-out"`
+	IfIdIn        string `vici:"if-id-in"`
+	IfIdOut       string `vici:"if-id-out"`
+	EncrAlg       string `vici:"encr-alg"`
+	EncKeysize    string `vici:"encr-keysize"`
+	IntegAlg      string `vici:"integ-alg"`
+	IntegKeysize  string `vici:"integ-keysize"`
+	DhGroup       string `vici:"dh-group"`
+	Esn           string `vici:"esn"`
+}
+
+type list_ike_sa struct {
+	UniqueId      string                   `vici:"uniqueid"`
+	Version       string                   `vici:"version"`
+	State         string                   `vici:"state"`
+	LocalHost     string                   `vici:"local-host"`
+	LocalPort     string                   `vici:"local-port"`
+	LocalId       string                   `vici:"local-id"`
+	RemoteHost    string                   `vici:"remote-host"`
+	RemotePort    string                   `vici:"remote-port"`
+	RemoteId      string                   `vici:"remote-id"`
+	RemoteXauthId string                   `vici:"remote-xauth-id"`
+	RemoteEapId   string                   `vici:"remote-eap-id"`
+	Initiator     string                   `vici:"initiator"`
+	InitiatorSpi  string                   `vici:"initiator-spi"`
+	ResponderSpi  string                   `vici:"responder-spi"`
+	NatLocal      string                   `vici:"nat-local"`
+	NatRemote     string                   `vici:"nat-remote"`
+	NatFake       string                   `vici:"nat-fake"`
+	NatAny        string                   `vici:"nat-any"`
+	IfIdIn        string                   `vici:"if-id-in"`
+	IfIdOut       string                   `vici:"if-id-out"`
+	EncrAlg       string                   `vici:"encr-alg"`
+	EncrKeysize   string                   `vici:"encr-keysize"`
+	IntegAlg      string                   `vici:"integ-alg"`
+	IntegKeysize  string                   `vici:"integ-keysize"`
+	PrfAlg        string                   `vici:"prf-alg"`
+	DhGroup       string                   `vici:"dh-group"`
+	Ppk           string                   `vici:"ppk"`
```

Comment:
I don't see this here, is there somewhere else I should be looking?  https://github.com/strongswan/strongswan/blob/master/src/libcharon/plugins/vici/README.md#list-sa

Reply:
Yes, the source code is our friend: https://github.com/strongswan/strongswan/blob/master/src/libcharon/plugins/vici/vici_query.c#L497

Which makes me think I should re-examine the connection stuff, and will do that next. Thanks again for the review!